.changeset/tricky-clocks-end.md

```diff
@@ -0,0 +1,7 @@
+---
+'@astrojs/vercel': major
+---
+
+The Vercel adapter now streams responses!
+
+This brings better performance to your visitors by showing them content as it is rendered. The browser can also start loading the required stylesheets and scripts much sooner, which ultimately results in faster full page loads.
```

Comment:
``````suggestion
This brings better performance to your visitors by showing them content as it is rendered. The browser can also start loading the required stylesheets and scripts much sooner, which ultimately results in faster full page loads.

This is a breaking change and may affect your project. Features that modify the [Response headers](https://developer.mozilla.org/en-US/docs/Glossary/Response_header) are only available at the page level when HTML streaming is enabled. By the time Astro runs your component code, it has already sent the Response headers and they cannot be modified.

HTML streaming can be disabled in the `vercel` adapter configuration:

```js
import { defineConfig } from 'astro/config';
import vercel from '@astrojs/vercel/serverless';

export default defineConfig({
  output: 'server',
  adapter: vercel(
  	streaming: false
  ),
});
```

Read more about [HTML streaming](https://docs.astro.build/en/guides/server-side-rendering/#html-streaming) in the Astro docs.
``````

What about something a little beefier for a major (or minor with breaking change) like this? (Edit as necessary, and show whatever config option can be set.

NOTE: Depending on how quickly this merges, I have a PR that would add the anchor link above but currently the URL is only `#streaming`. If this is to merge soon, I can pre-emptively update existing docs to that link now, even before the new page is published.